### ISSUE TITLE
Return NaN in Fast_Path Mode

### ISSUE THREAD

**Issue by kangshwan** [state-spaces/mamba#649]
**Dear Authors,**

Hello, I am currently utilizing Mamba to build a Diffusion Model. Following the Motion Mamba paper, I have structured my model, but I am encountering the following issue:

# Problem:
When implementing the Motion Mamba model following the paper, I encounter an issue where the model starts returning NaN values starting from the 32nd denoising step out of a total of 50 steps. This happens even though no training has been performed; it occurs during **sanity checks** in PyTorch Lightning.

Interestingly, this issue does not occur **when `use_fast_path` is disabled.**

# Environment:

- Docker container
- Ubuntu 20.04.6 LTS
- RTX A6000
- Python version: 3.9.20
- CUDA 11.8
- conv1d: 1.2.0.post2

# Question:
I have implemented the code blocks while preserving the structure of the original Mamba Block as closely as possible.
Could you provide insights or suggestions on why NaN values might be occurring, specifically in relation to `use_fast_path`?

# Code:
Below is the forward function I am using:
```
if self.use_fast_path and causal_conv1d_fn is not None and inference_params is None:  # Doesn't support outputting the states
    outputs = []
    for idx, (dt_proj, conv1d, A_log, D )in enumerate(zip(self.dt_projs, self.conv1ds, self.A_logs, self.Ds)):
        A = -torch.exp(A_log.float()) 
        out = mamba_inner_fn_no_out_proj(
            xz,
            conv1d.weight,
            conv1d.bias,
            self.x_proj.weight,
            dt_proj.weight,
            A,
            None,  # input-dependent B
            None,  # input-dependent C
            D.float(),
            delta_bias=dt_proj.bias.float(),
            delta_softplus=True,
        )
        out = rearrange(out, "b d l -> b l d")
        if np.isnan(out.cpu()).any():
            print(f"block no.{idx} produce Nan ")
        outputs.append(out)

    if self.aggregate == 'sum':
        aggregated = torch.stack(outputs, dim=0).sum(dim=0)
    elif self.aggregate == 'concat_linear':
        aggregated = torch.cat(outputs, dim=-1)
        aggregated = self.aggregate_linear(aggregated)

    out = self.out_proj(aggregated)

    return out
else:
    x, z = xz.chunk(2, dim=1)
    outputs = []
    for dt_proj, conv1d, A_log, D in zip(self.dt_projs, self.conv1ds, self.A_logs, self.Ds):
        A = -torch.exp(A_log.float())  # (d_inner, d_state)

        # Compute short convolution
        if conv_state is not None:
            # If we just take x[:, :, -self.d_conv :], it will error if seqlen < self.d_conv
            # Instead F.pad will pad with zeros if seqlen < self.d_conv, and truncate otherwise.
            conv_state.copy_(F.pad(x, (self.d_conv - x.shape[-1], 0)))  # Update state (B D W)
        if causal_conv1d_fn is None:
            x = self.act(conv1d(x)[..., :seqlen])
        else:
            assert self.activation in ["silu", "swish"]
            x = causal_conv1d_fn(
                x=x,
                weight=rearrange(conv1d.weight, "d 1 w -> d w"),
                bias=conv1d.bias,
                activation=self.activation,
            )
        # We're careful here about the layout, to avoid extra transposes.
        # We want dt to have d as the slowest moving dimension
        # and L as the fastest moving dimension, since those are what the ssm_scan kernel expects.
        x_dbl = self.x_proj(rearrange(x, "b d l -> (b l) d"))  # (bl d)
        dt, B, C = torch.split(x_dbl, [self.dt_rank, self.d_state, self.d_state], dim=-1)
        dt = dt_proj.weight @ dt.t()
        # Paper: N, code: d_state
        dt = rearrange(dt, "d (b l) -> b d l", l=seqlen)
        B = rearrange(B, "(b l) dstate -> b dstate l", l=seqlen).contiguous()
        C = rearrange(C, "(b l) dstate -> b dstate l", l=seqlen).contiguous()
        assert self.activation in ["silu", "swish"]
        y = selective_scan_fn(
            x,
            dt,
            A,
            B,
            C,
            D.float(),
            z=z,
            delta_bias=dt_proj.bias.float(),
            delta_softplus=True,
            return_last_state=ssm_state is not None,
        )
        if ssm_state is not None:
            y, last_state = y
            ssm_state.copy_(last_state)
        y = rearrange(y, "b d l -> b l d")
        outputs.append(y)

    if self.aggregate == 'sum':
        aggregated = torch.stack(outputs, dim=0).sum(dim=0)
    elif self.aggregate == 'concat_linear':
        aggregated = torch.cat(outputs, dim=-1)
        aggregated = self.aggregate_linear(aggregated)

    out = self.out_proj(aggregated)
    return out
```

# Reference Code:
I referred to the following for sub-functions:
https://github.com/hustvl/Vim/blob/main/mamba-1p1p1/mamba_ssm/ops/selective_scan_interface.py

Thank you for your time and assistance!

Best regards,
Seunghwan Kang

**Comment by kangshwan** [comment]
Sry for my misunderstanding.
I think this is cuz normalization layer is not in my code.
